scripts/scr_poststage.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+#
+# SCR allows you to spawn off dataset transfers "in the background"
+# that will finish some time after a job completes.  This saves you from
+# using your compute time to transfer datasets.  You can do this by
+# specifying XFER=BBAPI_POSTSTAGE for a storage descriptor in scr.conf.
+# Currently, this is only supported on on IBM burst buffer nodes.
+#
+# This script is to be run as a 2nd-half post-stage script on an IBM
+# system.  A 2nd-half post-stage script will run after all the job's burst
+# buffers transfers have finished (which could be hours later after the job
+# finishes).  This script will finalize any completed burst buffer dataset
+# transfers so that they're visible to SCR.
+#
+
+# Please fill in BINDIR AND PREFIX below:
+
+# The full path to the dir containing the SCR binaries (scr_index and scr_flush_file)
+BINDIR=""
+
+# The dir path to your SCR PREFIX
+PREFIX="/p/gpfs1/`whoami`"
+
+# Path to where you want the scr_poststage.sh log
+LOGFILE=$PREFIX/scr_poststage.log
+
+if [ ! -d "$BINDIR" ] ; then
+	echo "Please specify BINDIR in scr_poststage.sh" >> $LOGFILE
+	exit
+fi
+
+if [ ! -d "$PREFIX" ] ; then
+	echo "Please specify PREFIX in scr_poststage.sh" >> $LOGFILE
+	exit
+fi
+
+export PATH="$PATH:$BINDIR"
+
+function do_poststage {
+	echo "$(date '+%m/%d/%y %H:%m:%S') Begin post_stage" >> $LOGFILE
+	echo "Current index before finalizing:" >> $LOGFILE
+	scr_index -l -p $PREFIX >> $LOGFILE
+
+	# Finalize each dataset listed in the flush file
+	for ID in $(scr_flush_file --dir $PREFIX --list-ckpt) ; do
+		DSET="$(scr_flush_file --dir $PREFIX --name $ID)"
+
+		echo "Looking at dataset $ID ($DSET)" >> $LOGFILE
+
+		if ! scr_flush_file --dir $PREFIX --need-flush $ID &>> $LOGFILE ; then
+			# Dataset is already flushed, skip it
+			continue
+		fi
+
+		echo "Finalizing transfer for dataset $ID ($DSET)" >> $LOGFILE
+		if ! scr_flush_file -r -s $ID --dir $PREFIX &>> $LOGFILE ; then
+			echo "Error: Can't resume dataset $ID, rc=$?" >> $LOGFILE
+			continue
+		fi
+
+		echo "Writing summary for dataset $ID" >> $LOGFILE
+		if ! scr_flush_file -s $ID -S --dir $PREFIX &>> $LOGFILE ; then
+			echo "ERROR: Can't write summary for dataset $ID" >> $LOGFILE
+			continue
+		fi
+
+		echo "Adding dataset $DSET to index" >> $LOGFILE
+		if ! scr_index -p $PREFIX --add=$DSET &>> $LOGFILE ; then
+			echo "Couldn't add dataset $DSET to index" >> $LOGFILE
+			continue
+		fi
+
+		echo "Setting current dataset to $DSET" >> $LOGFILE
+		if ! scr_index -p $PREFIX --current=$DSET &>> $LOGFILE ; then
+			echo "Couldn't set current dataset to $DSET" >> $LOGFILE
+			continue
+		fi
```

Comment:
I took a look at the code.  I think ``scr_index`` will avoid setting a non-checkpoint to be current.  However, it also will print a message and return with an error:

https://github.com/LLNL/scr/blob/55c800d60d621d538c48d1db2cc692a07477273a/src/scr_index.c#L2022

Let's avoid printing any error messages so that we don't throw a false alarm for the user.  You can use ``test_api`` to write a non-checkpoint by adding an ``--output`` option.  I think ``--output 1`` will write every dataset as pure output (non-checkpoint).